jerry-core/ecma/builtin-objects/ecma-builtin-reflect.c

```diff
@@ -227,19 +228,36 @@ ecma_builtin_reflect_dispatch_routine (uint16_t builtin_routine_id, /**< built-i
     }
     case ECMA_REFLECT_OBJECT_SET_PROTOTYPE_OF:
     {
-      ecma_value_t result = ecma_builtin_object_object_set_prototype_of (arguments_list[0], arguments_list[1]);
-      bool is_error = ECMA_IS_VALUE_ERROR (result);
+      if (ECMA_IS_VALUE_ERROR (ecma_op_check_object_coercible (arguments_list[0])))
+      {
+        return ECMA_VALUE_ERROR;
+      }
 
-      if (is_error)
+      if (!ecma_is_value_object (arguments_list[1]) && !ecma_is_value_null (arguments_list[1]))
       {
-        jcontext_release_exception ();
+        return ecma_raise_type_error (ECMA_ERR_MSG ("proto is neither Object nor Null."));
+      }
+
+      if (!ecma_is_value_object (arguments_list[0]))
+      {
+        return ecma_copy_value (arguments_list[0]);
```

Comment:
This should be a TypeError according to the standard. the order of the checks should be swapped.